From 857e1d708ac5f51253bfeedf7634515d4bde0bd9 Mon Sep 17 00:00:00 2001
From: Mathieu Poirier <mathieu.poirier@linaro.org>
Date: Fri, 4 Jul 2025 15:13:55 -0600
Subject: [PATCH] coco: guest: arm64: Don't perform MMIO range validation

On QEMU, the SPDM responder and TDISP functionality are provided
an external library.  As such, things like BAR addresses are not
communicated with the DSM, making their validation impossible.

Signed-off-by: Mathieu Poirier <mathieu.poirier@linaro.org>
---
 drivers/virt/coco/arm-cca-guest/rsi-da.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/drivers/virt/coco/arm-cca-guest/rsi-da.c b/drivers/virt/coco/arm-cca-guest/rsi-da.c
index f052dffb35b1f..2f81864bd5370 100644
--- a/drivers/virt/coco/arm-cca-guest/rsi-da.c
+++ b/drivers/virt/coco/arm-cca-guest/rsi-da.c
@@ -427,7 +427,9 @@ int rsi_device_start(struct pci_dev *pdev)
 					       mmio_start_phys, mmio_flags);
 		if (ret) {
 			pci_err(pdev, "failed to set protection attributes for the address range\n");
+#if 0
 			return -EIO;
+#endif
 		}
 	}
 
-- 
GitLab

